Разработаны почти все тесты по TVector'у

McNikidry opened the pull request.

McNikidry committed:
Add files via upload

McNikidry committed:
Разработаны почти все тесты по TVector'у

AlexSysoyev commented on test/test_tvector.cpp:
```diff
@@ -31,7 +31,13 @@ TEST(TVector, can_create_copied_vector)
+	int SI = 0;
+	TVector<int> one(size, SI);
+
+	TVector<int> two = one;
```
надо изменить в векторах элементы, а уже потом сравнивать.

AlexSysoyev commented on test/test_tvector.cpp:
```diff
@@ -63,91 +69,269 @@ TEST(TVector, can_set_and_get_element)
+	int size = 4; int size1 = 5; int si = 0;
+
+	TVector<int> tmp(size, si);
+	TVector<int> tmp2(size1, si);
```
Не так. Нужно проверить, что верно отработает tmp = tmp

AlexSysoyev commented on test/test_tvector.cpp:
```diff
@@ -63,91 +69,269 @@ TEST(TVector, can_set_and_get_element)
+
+	TVector<int> tmp(size, si);
+	TVector<int> tmp1(size, si);
+
```
Где присваивание?

AlexSysoyev commented on test/test_tvector.cpp:
```diff
@@ -63,91 +69,269 @@ TEST(TVector, can_set_and_get_element)
+
+	TVector<int> tmp(size, si);
+	TVector<int> tmp1(size1, si);
+
```
Где присваивание?

McNikidry committed:
Исправлены ошибки  и добавлены тесты.

Возникло пару вопросов, что значит оператор изменения размера вектора в тестах?

alvls commented on test/test_tvector.cpp:
```diff
@@ -63,91 +72,293 @@ TEST(TVector, can_set_and_get_element)
+
+	tmp = tmp;
+
+	EXPECT_EQ(tmp, tmp);
```
Здесь лучше проверять на отсутствие исключений. Они могут возникнуть при неверной реализации.

alvls commented on test/test_tvector.cpp:
```diff
@@ -63,91 +72,293 @@ TEST(TVector, can_set_and_get_element)
+	TVector<int> tmp(size, si);
+	TVector<int> tmp1(size, si);
+
+	tmp = tmp1;
```
Вектора были одинаковыми и до присваивания, "по построению". Исправьте

alvls commented on test/test_tvector.cpp:
```diff
@@ -63,91 +72,293 @@ TEST(TVector, can_set_and_get_element)
+
+	three = two + one;
+
+	tmp[0] = 4;
```
Неверный тест. Сложение векторов разных размеров должно быть невозможно.

McNikidry replied:
Т.е мне делать тест на ловлю throw?

alvls replied:
да

alvls commented on test/test_tvector.cpp:
```diff
@@ -63,91 +72,293 @@ TEST(TVector, can_set_and_get_element)
+
+	three = two - one;
+
+	tmp[0] = 3;
```
Аналогично тесту на  сложение

alvls commented on test/test_tvector.cpp:
```diff
@@ -63,91 +72,293 @@ TEST(TVector, can_set_and_get_element)
+
+	res = two * one;
+
+	tmp = 8;
```
Аналогично тесту на сложение.

McNikidry committed:
Исправил тесты

McNikidry committed:
Доделал тесты, исправил ошибки

alvls commented on test/test_tvector.cpp:
```diff
@@ -1,153 +1,326 @@
+
+	TVector<int> two = one;
+
+	two[2] = 2;
```
Неверно. Сначала нужно поменять значения в one, потом скопировать, потом проверять.

alvls commented on test/test_tvector.cpp:
```diff
@@ -1,153 +1,326 @@
+	TVector<int> one(size, si);
+	TVector<int> two(size, si);
+
+	EXPECT_EQ(one, two);
```
Сначала измените вектора

McNikidry committed:
Исправил
